auth/realm/base/src/main/java/org/wildfly/security/auth/realm/ObjectCallback.java

```diff
@@ -0,0 +1,40 @@
+package org.wildfly.security.auth.realm;
+
+import javax.security.auth.callback.Callback;
+
+/**
+ * An implementation of Callback that simply obtains an Object to be used
+ * as the authentication credential. Interpretation of the Object is up to
+ * the LoginModules that validate the credential.
+ *
+ * @author Scott.Stark@jboss.org
+ * @version $Revision$
+ */
+public class ObjectCallback implements Callback {
```

Comment:
I think supporting this callback may be worth discussing more, as a new callback existing login modules will not be using this - maybe for now we would be better sticking with the PasswordCallback and CredentialCallback support - at least the former of the two should be a standard type?